DrawingBox/DrawingBox/Box.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+
+namespace DrawingBox
+{
+    public static class Box
+    {
+        private static int x;
+        private static int y;
+        private static int height;
+        private static int width;
+        private static char borderSymbol;
+        private static string message;
+
+        public static readonly char[] ValidCharacters;
+
+        private static void draw()
+        {
+            var bufferHeight = Console.BufferHeight;
+            var bufferWidth = Console.BufferWidth;
+
+            if (Box.y + Box.height > bufferHeight)
+                throw new InvalidOperationException($"(top offset + height) cannot be more than buffer height: {bufferHeight}");
+
+            if (Box.x + Box.width > bufferWidth)
+                throw new InvalidOperationException($"(left offset + width) cannot be more than buffer width: {bufferWidth}");
+
+            var borderSymbolIsValid = false;
+            foreach (var symbol in Box.ValidCharacters)
+            {
+                if (Box.borderSymbol == symbol)
+                    borderSymbolIsValid = true;
+            }
+
+            if (!borderSymbolIsValid)
+                throw new InvalidOperationException($"You're not allowed to use border symbol different from this: [ *, +, . ]");
+
+            if (Box.height < 4 && Box.width < 4)
+                throw new InvalidOperationException("You cannot make box with width and height less than 4");
+
+            Console.Clear();
+
+            Box.drawBorder();
+            Box.drawMessage();
+        }
+        private static void drawBorder()
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+
+            Console.SetCursorPosition(Box.x, Box.y);
+            Console.WriteLine(new string(Box.borderSymbol, Box.width));
+
+            Console.SetCursorPosition(Box.x, Box.y + Box.height - 1);
+            Console.WriteLine(new string(Box.borderSymbol, Box.width));
+
+            for (int row = 1; row < Box.height - 1; row++)
+            {
+                Console.SetCursorPosition(Box.x, Box.y + row);
+                Console.Write(Box.borderSymbol);
+
+                Console.SetCursorPosition(Box.x + Box.width - 1, Box.y + row);
+                Console.Write(Box.borderSymbol);
+            }
+
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+        private static void drawMessage()
+        {
+            Console.ForegroundColor = ConsoleColor.Yellow;
+
+            var messageArea = (Box.height - 2) * (Box.width - 2);
```

Comment:
(Box.height - 2) and (Box.width - 2)
You repeat these calculations couple times. I think better performance way to declare 2 new variables, calculate these values once and use they during executing methods.

Reply:
Fixed in commit 3246ffc9de4f9d210989bb2bd682c40eebcd0e94